libs/components/forms/src/lib/modules/input-box/input-box.component.ts

```diff
@@ -78,38 +147,59 @@ export class SkyInputBoxComponent implements OnInit {
   @ContentChild(NgModel)
   public ngModel: NgModel | undefined;
 
-  public get hasErrorsComputed(): boolean {
+  @ContentChild(SkyInputBoxControlDirective, {
+    read: ElementRef,
+  })
+  public inputRef: ElementRef | undefined;
+
+  protected controlDir: AbstractControlDirective | undefined;
+  protected helpPopoverOpen: boolean | undefined;
+
+  protected get hasErrorsComputed(): boolean {
     if (this.hasErrors === undefined) {
-      return !!(
-        this.#controlHasErrors(this.formControl) ||
-        this.#controlHasErrors(this.formControlByName) ||
-        this.#controlHasErrors(this.ngModel)
-      );
+      return this.#controlHasErrors(this.controlDir);
     }
 
     return this.hasErrors;
   }
 
-  #changeRef: ChangeDetectorRef;
-  #inputBoxHostSvc: SkyInputBoxHostService;
-  #adapterService: SkyInputBoxAdapterService;
-  #elementRef: ElementRef;
-  constructor(
-    changeRef: ChangeDetectorRef,
-    inputBoxHostSvc: SkyInputBoxHostService,
-    adapterService: SkyInputBoxAdapterService,
-    elementRef: ElementRef
-  ) {
-    this.#changeRef = changeRef;
-    this.#inputBoxHostSvc = inputBoxHostSvc;
-    this.#adapterService = adapterService;
-    this.#elementRef = elementRef;
+  protected get required(): boolean {
+    return (
+      this.controlDir?.control?.hasValidator(Validators.required) ||
+      this.inputRef?.nativeElement.required
+    );
   }
 
+  #_stacked = false;
+  #_characterLimit: number | undefined;
+
+  #previousInputRef: ElementRef | undefined;
+  #previousMaxLengthValidator: ValidatorFn | undefined;
+
   public ngOnInit(): void {
     this.#inputBoxHostSvc.init(this);
   }
 
+  public ngAfterContentChecked(): void {
+    this.controlDir =
+      this.formControl || this.formControlByName || this.ngModel;
+
+    if (this.inputRef && this.inputRef !== this.#previousInputRef) {
+      const el = this.inputRef.nativeElement;
+
+      this.#renderer.addClass(el, 'sky-form-control');
+      this.#renderer.setAttribute(el, 'aria-describedby', this.errorId);
```

Comment:
I think we should also be doing this for aria required? 🤔 right now it isn't announcing if the field is required.

Reply:
Fixed. I also added `aria-invalid` for when the field has errors.